pkg/tool/common.go

```diff
@@ -27,8 +27,8 @@ type LogQL struct {
 
 type Checker interface {
 	Transform(arg string, matchers *map[string]string) (string, error)
-	ValidateRules(data []byte) (*rulefmt.RuleGroups, error)
-	ValidateConfig(filename string) (error)
+	ValidateRules(filename string, data []byte) (*rulefmt.RuleGroups, error)
```

Comment:
Seems a bit redundant to pass the name of the file *and* it's contents. Could we just read the file inside this function?

Reply:
im torn as i also see the benefit of having a function named `ValidateSomething` not have the side-effect of also having disk IO, but on the other hand i don't think it matters much.